crates/router/src/core/user.rs

```diff
@@ -1766,3 +1768,49 @@ pub async fn generate_recovery_codes(
         recovery_codes: recovery_codes.into_inner(),
     }))
 }
+
+pub async fn terminate_2fa(
+    state: AppState,
+    user_token: auth::UserFromSinglePurposeToken,
+    skip_2fa: Option<bool>,
+) -> UserResponse<user_api::TokenResponse> {
+    let user_from_db: domain::UserFromStorage = state
+        .store
+        .find_user_by_id(&user_token.user_id)
+        .await
+        .change_context(UserErrors::InternalServerError)?
+        .into();
+
+    if skip_2fa.is_none() {
+        if !(check_totp_in_redis(&state, &user_token.user_id).await?
+            || check_access_code_in_redis(&state, &user_token.user_id).await?)
```

Comment:
```suggestion
        if !check_totp_in_redis(&state, &user_token.user_id).await?
            && !check_access_code_in_redis(&state, &user_token.user_id).await?
```
This is more readable imo.

Reply:
Changed